### DISCUSSION TITLE
[Bug]: Deck suggestion is not working.

### DISCUSSION THREAD

**Discussion by cundas** [unrealities/MTGA_Draft_17Lands#68] [Ideas]
### Describe the bug in detail

Everything works well , i also dont click before the draft pick end but still deck suggestion is not working , how i can fix it? 

### Steps to reproduce this bug

Start draft , end pick time and then click the button "Suggest Deck"

### Operating System

Windows

### Operating System Version

Windows 11

### Run Type

Windows Executable

**Comment by unrealities** [comment]
The suggest deck works best for two-color decks. I've found that if I'm playing three or more, it won't give a suggestion because I don't have enough cards for a pure two-color deck.

**Comment by FiYir** [comment]
I drafted a bunch of mediocre cards early in DSK, and I also wasn't getting any results from the suggester. I was able to get suggestions once I lowered the `ratings_threshold` in config.json from 500 to 100, restarted the app, and opened the draft log.

![image](https://github.com/user-attachments/assets/421180e2-e90a-447e-92f5-a4058d4d3b2a) 
![image](https://github.com/user-attachments/assets/f77e6a4f-7593-49ea-9f52-4a6ff07fd925)

**Comment by unrealities** [comment]
That is interesting, and it is a reasonable deck. I would like to know if the data for the color pairs isn't the best after the first day. I've been drafting using the tool for the new set and haven't seen anything strange. I did sealed the first day before data was available, so I don't see your other issue that I just pushed.

**Comment by FiYir** [comment]
@unrealities, I added my sealed log entry to test_log_scanner so you can check if the format matches the entry from your event. The entry may change depending on the number of active events (for example, 'Course' may change to 'Courses'). It’s probably a good idea to open your sealed log in version 3.27 and consider adding your entry to test_log_scanner to confirm that my change didn’t break anything.

**Comment by brobson** [comment]
It looks like the deck suggestion is ignoring Room cards, likely related to this

Nevermind, looks like they are included in suggestions.  Maybe since the 3.27 update?

**Comment by FiYir** [comment]
@brobson, version 3.27 didn't change anything regarding the suggestions; it only fixed an issue with the sealed event.

I haven't had any issues receiving suggestions that included room cards, but I've also been using the Premier Draft, All Groups dataset for all events. Is this what you're using?

**Comment by brobson** [comment]
@FiYir I have been using Premier Draft, All Groups as well.

I was thinking that the rooms might be determined by their total mana value and so they are lower weighted due to curve calculations.  Meat Locker // Drowned Diner, Underwater Tunnel // Slimy Aquarium, and Grand Entryway // Elegant Rotunda were not being recommended for my White/Blue decks.  

I edited my post to add the second line about 3.27 because right after I commented here I had Glassworks // Shattered Yard and Ticket Booth // Tunnel of Hate suggested for my Green/Red deck.  It seems to be that the rooms are just lower winrate than I though for White/Blue, kicking them out of the suggestions.

**Comment by unrealities** [comment]
Improvements are needed for high CMC cards that can be played at lower costs, including overlords.
* Overlord of the Mistmoors (68.9% GIHWR) has a CMC of 7 but can be played for 4.
* Unholy Annex // Ritual Chamber (65.6% GIHWR) has a CMC of 8 but can be played for 3.

Here's a snippet of Scryfall JSON for Unholy Annex:
```json
{
  "name": "Unholy Annex // Ritual Chamber",
  "mana_cost": "{2}{B} // {3}{B}{B}",
  "cmc": 8.0
}
```

Also, recently, many non-creature spells are producing bodies. So not only is this card excluded because of its high CMC, but it is also not considered a creature. This card is a CMC 5 creature and a CMC 3 enchantment.

We can modify the code to parse `mana_cost` instead of `cmc`, but accounting for it being a creature seems much more difficult.
